lib/document/stateTransition/DocumentsStateTransition.js

```diff
@@ -1,14 +1,30 @@
 const AbstractStateTransition = require('../../stateTransition/AbstractStateTransition');
 const stateTransitionTypes = require('../../stateTransition/stateTransitionTypes');
 
+const AbstractDocumentTransition = require('./documentTransition/AbstractDocumentTransition');
+const DocumentCreateTransition = require('./documentTransition/DocumentCreateTransition');
+const DocumentReplaceTransition = require('./documentTransition/DocumentReplaceTransition');
+const DocumentDeleteTransition = require('./documentTransition/DocumentDeleteTransition');
+
+const actionsToClasses = {
+  [AbstractDocumentTransition.ACTIONS.CREATE]: DocumentCreateTransition,
+  [AbstractDocumentTransition.ACTIONS.REPLACE]: DocumentReplaceTransition,
+  [AbstractDocumentTransition.ACTIONS.DELETE]: DocumentDeleteTransition,
+};
+
 class DocumentsStateTransition extends AbstractStateTransition {
```

Comment:
I would rename it in order to remove ambiguity with the document transition term. What do you think about `DocumentsBatchTransition`?